tools/clang/tools/dxcompiler/dxcutil.cpp

```diff
@@ -177,23 +149,10 @@ HRESULT ValidateAndAssembleToContainer(AssembleInputs &inputs) {
   std::unique_ptr<llvm::Module> llvmModuleWithDebugInfo;
 
   CComPtr<IDxcValidator> pValidator;
-  bool bInternalValidator = CreateValidator(pValidator, inputs.SelectValidator);
-  // Warning on internal Validator
-
-  CComPtr<IDxcValidator2> pValidator2;
-  if (!bInternalValidator) {
-    pValidator.QueryInterface(&pValidator2);
-  }
+  CreateValidator(pValidator);
 
-  if (bInternalValidator || pValidator2) {
-    // If using the internal validator or external validator supports
-    // IDxcValidator2, we'll use the modules directly. In this case, we'll want
-    // to make a clone to avoid SerializeDxilContainerForModule stripping all
-    // the debug info. The debug info will be stripped from the orginal module,
-    // but preserved in the cloned module.
-    if (llvm::getDebugMetadataVersionFromModule(*inputs.pM) != 0) {
-      llvmModuleWithDebugInfo.reset(llvm::CloneModule(inputs.pM.get()));
-    }
+  if (llvm::getDebugMetadataVersionFromModule(*inputs.pM) != 0) {
+    llvmModuleWithDebugInfo.reset(llvm::CloneModule(inputs.pM.get()));
   }
 
   // Verify validator version can validate this module
```

Comment:
I think the following section of code checking the validator version may also be unnecessary, since we shouldn't allow selection of a newer validator version than supported by the compiler.